cpp/tests/CMakeLists.txt

```diff
@@ -14,38 +14,58 @@
 # limitations under the License.
 #=============================================================================
 
-if(NOT CMAKE_CUDA_COMPILER)
-  message(SEND_ERROR "CMake cannot locate a CUDA compiler")
+cmake_minimum_required(VERSION 3.14...3.18 FATAL_ERROR)
+
+project(CUDF_TESTS VERSION 0.18.0 LANGUAGES C CXX CUDA)
```

Comment:
Can remove these I believe

Reply:
Without this, CMake prints a dev warning about the files not calling `project()`

Reply:
hmm, we were able to remove this in RMM without issue: https://github.com/rapidsai/rmm/blob/branch-0.18/tests/CMakeLists.txt

Reply:
The `project()` calls define vars like `CUDF_TEST_SOURCE_DIR` and `CUDF_BENCHMARKS_SOURCE_DIR`.

Reply:
We should be able to use `CMAKE_CURRENT_SOURCE_DIR` instead of those: https://github.com/rapidsai/rmm/blob/branch-0.18/tests/CMakeLists.txt

Reply:
Yes but does that mean we _should_? :upside_down_face: